Migrated TFS 2010 Bug Reporting Guide (#13)
esdc-devcop/guides: source-control/Bug_Reporting_TFS_2010.md

    * [1.4.11 Details:](#.2ADetails:)
    * [1.4.12 Test Cases:](#Test_Cases:)
    * [1.4.13 All Links:](#All_Links:)
    * [1.4.14 Attachments:](#Attachments:)
  * [1.5 Bug Reporting Process](#Bug_Reporting_Process)
    * [1.5.1 Reporting the Bug](#Reporting_the_Bug)
    * [1.5.2 Resolving the Bug](#Resolving_the_Bug)
    * [1.5.3 Building and Deploying the new version](#Building_and_Deploying_the_new_version)
    * [1.5.4 Validating the Resolution](#Validating_the_Resolution)
  * [1.6 Bug Reporting Diagram](#Bug_Reporting_Diagram)
  * [1.7 Team Queries](#Team_Queries)
    * [1.7.1 Active Bugs](#Active_Bugs)
    * [1.7.2 Bugs to be tested and/or closed](#Bugs_to_be_tested_and.2For_closed)
  * [1.8 Build Definition Naming Convention (.Net Stack)](#Build_Definition_Naming_Convention_.28.Net_Stack.29)
  * [1.9 Testing Complex or Multi-Application Solutions](#Testing_Complex_or_Multi-Application_Solutions)
    * [1.9.1 Option 1: Provide access to your TFS](#Option_1:_Provide_access_to_your_TFS)
    * [1.9.2 Option 2: Merge into 1 Team Project](#Option_2:_Merge_into_1_Team_Project)
    * [1.9.3 Option 3: Create a Central Team Project](#Option_3:_Create_a_Central_Team_Project)

>   * [1.10 Template for "Details = Steps to Reproduce" Field](#Template_for_.22Details_.3D_Steps_to_Reproduce.22_Field)

Reviewer: Not all these links work. I don't think we need all of them either; probably just the first level.

This person is usually a member of the Development Team, or the Product Manager.

#### TST Coordinator

The TST Coordinator is usually a member of the Testing Team.  
Their role is to analyze the Resolved bugs and assign them to the proper Validator for testing.

#### Validator

The Validator is the person that will test the resolution and determine if the issue is properly resolved.  
The Validator is usually the same person as the Reporter of the issue but can be another person.  
The Validator will set the bug to the Closed state.

### Fields of a Bug

The fields available in the Bug work item type have been designed to help with tracking, reporting and to collaborate efficiently. To document the defect, the fields need to be properly populated and can be updated during the bug reporting process as details of the defect unfold. Incorrect values can impact the results of queries, which are used to list and assign work to individuals.

Below are the recommendations for the main fields of the Bug Work Item. Fields marked with an  are the most crucial to populate.

> [![image](/gcwiki/images/f/f7/BugWindow.JPG)](/wiki/File:BugWindow.JPG "image")

Reviewer: The images aren't loading. You'll want to upload them to the `assets` folder.

#### Title:

* must be short, simple and precise to the issue found
* must have meaning
* allows you to distinguish from other bugs
* is easy to identify the source of the problem (e.g., include page, or action)
* should not contain details found in other fields (area, iteration, build, environment...)
* If the is bug generated from an external source (help desk), the tracking number used by that tool should be included in the title. (ex: SRMIS #)

    Examples:
    Registration - Search field label reads "Period"; should read "Period Submitted"
    Edit Contact - Error raised with valid postal code
    SRMIS* 1273664 - User cannot log in
    Search - possible SQL Injection

#### Assign To:

* Identifies who in the project team is responsible for taking action on the work item.
* Generic lists should be used when assigning to the Dev Coordinator and TST Coordinator.
* The Dev and TST Coordinators will select the specific resource once it's decided who will action the work item.

#### States and Reasons

Below you will find the States and their available Reasons for the work item type Bug in the default Agile Template.

> [![image](/gcwiki/images/f/f6/BugReasonsStates2.JPG)](/wiki/File:BugReasonsStates2.JPG "image")

Reviewer: Image not loading. Should upload it to `assets` folder.

#### Area

* Identifies the application, component or module impacted
* It can be modified when the resolution of the bug is found. Example: the area selected at the creation of the Bug was "MSCBA Web" but after investigation the issue was in the Stored Procedure and the area was modified to reflect this.
* Do not use Area to group and/or track bugs raised by Security/Performance testing. The field "Classification" should be used for this.

> * Guidelines can be found in the \[[TFS Project Setup Recommendation](Bug_Reporting_TFS_2010.md\]

Reviewer: This pointed to a dead link, probably just remove it.
```suggestion
```

#### Iteration:

* Iterations are a way to identify the cycle/release/milestone/sprint you are working on.

> * Guidelines can be found in the \[[TFS Project Setup Recommendation](Bug_Reporting_TFS_2010.md\]

Reviewer: This pointed to a dead link, probably just remove it.
```suggestion
```

1. Critical: Must fix. A defect that causes termination of one or more system components or the complete system, or causes extensive data corruption. And, there are no acceptable alternative methods to achieve required results.
2. High: Consider fix. A defect that causes termination of one or more system components or the complete system, or causes extensive data corruption. However, an acceptable alternative method exists to achieve required results.
3. Medium: (Default) A defect that causes the system to produce incorrect, incomplete or inconsistent results.
4. Low: A minor or cosmetic defect that has acceptable workarounds to achieve required results.

#### System Info:

* Tells us in which Build a bug was found and in which Build the correction was made.
* These fields are used by the Queries to determine which bugs are ready for testing. Useful when a new deploy of the application is made.
* The values for the "Found In Build" and "Integrated In Build" fields should only be selected from their respective drop downs.
  * These fields are automatically populated when the Build Server is used.

>   * Optionally they can be populated manually via the Global List. \[[Instructions to populate manually](Bug_Reporting_TFS_2010.md\].

Reviewer: Another previously dead link.
```suggestion
  * Optionally they can be populated manually via the Global List.
```

Reviewer: This dead link is still present.

       We  execute another build, Build Number "ABC\_MyCompoenent\_20160324.3" is created
       Application version is now at "2.3.3405"
       We release this version (2.3.3405) to systest, and this version includes changes from build "ABC\_MyComponent\_20160324.2" and "ABC\_MyComponent\_20160324.3"

#### Validating the Resolution

* Using the Team Query created in the previous step, the TST Coordinator is notified of the bugs which are ready to be tested or to be closed.
* The "Assign To" field is modified by the TST Coordinator with the proper Validator's name.
* The Validator executes its tests
* A decision is made to action the Bug.
* If the bug is to be closed or is resolved correctly:
  * The Status is changed to "Closed"
  * The cycle is complete
* If it is not resolved correctly:
  * The Status is changed back to "Active"
  * More details are provided to explain why it was rejected.
  * Fields can be updated if necessary
  * The cycle starts over.
* In all situations, all the Fields should be revised and updated if needed.

### Bug Reporting Diagram

[![Bugprocesstop.JPG](/gcwiki/images/thumb/1/11/Bugprocesstop.JPG/300px-Bugprocesstop.JPG)](/wiki/File:Bugprocesstop.JPG)

> [![Bugprocessbottom.JPG](/gcwiki/images/thumb/8/80/Bugprocessbottom.JPG/300px-Bugprocessbottom.JPG)](/wiki/File:Bugprocessbottom.JPG)

Reviewer: Images not loading. Should upload it to `assets` folder.

### Team Queries

The following "Team Queries" are executed to produce a list of the "Active Bugs" or Bugs that are "Ready for testing" for a specific "Iteration".

For future releases (Iterations), the queries can be copied and then the "Iteration" filter updated. They can also be copied by team members to their "My Queries" section and add a clause to specify work items assigned to them.

More filters could be added to better meet your needs example: The "Area" could be added if you have multiple components released in this same iteration and need to isolate the work items.

> Visit our "[TFS Project Setup Recommendation](Bug_Reporting_TFS_2010.md" for more details on Team Queries and naming convention.

Reviewer: Point to updated/migrated doc
```suggestion
Visit our [TFS 2015 Use and Configuration Recommendation](https://esdc-devcop.github.io/recommendations/tools/tfs2015.html) for more details on Team Queries and naming convention.
```

Reviewer: This link still isn't updated.

#### Active Bugs

This query will be executed mostly by the Dev Coordinator to list new bugs and bugs that have been re-activated for a particular iteration.

> [![image](/gcwiki/images/7/71/ActiveBugs_Query.JPG)](/wiki/File:ActiveBugs_Query.JPG "image")

Reviewer: Image not loading. Should upload it to assets folder.

#### Bugs to be tested and/or closed

This query will be executed mostly by the TST Coordinator to list bugs that have been resolved and are ready to be tested or be closed. At each deploy to the SysTest vertical, the clause "and Integration Build <=" will have to be modified to include the latest build deployed.

> [![image](/gcwiki/images/7/73/ReadytotestBugs_Query.JPG)](/wiki/File:ReadytotestBugs_Query.JPG "image")

Reviewer: Image not loading. Should upload it to assets folder.

* Bugs can be easily assigned and/or transferred to the other dev teams
* If a bug touches multiple systems, the original bug becomes the parent bug and new bugs, linked to the parent bug, are created for each impacted system. Only once all child bugs are closed can the parent bug can be closed.
* The Dev Coordinator and TST Coordinator play a more crucial role in analyzing and filtering the bugs and ensuring the fields are properly set and the bug is assigned to the proper team for resolution and/or for validation.

### Template for "Details = Steps to Reproduce" Field

Below is a template that should be used to populate the field "Steps to Reproduce" under the "Details" Tab.

   **PROBLEM**
   A short summary of the bug (i.e., 1 or 2 sentences)

   **URL**: URL where the issue was found

   **Environment**: Environment where the issue was found (TST, UAT...)
   **Date of Occurrence:** If different then the date the bug was created
   **USER:** Include user that was used when this bug was found and, if pertinent; include information on user role and privileges.  Remove this section
   if not applicable.

>    **Test Script:** If applicable list the automated test used to identify or reproduce the bug

Reviewer: Not Blocking: These are being bundled into a single paragraph, I'm not sure if that was intend or not.

Author: As intended.
